<ui version="4.0" >
 <class>RunSettings</class>
 <widget class="QWidget" name="RunSettings" >
  <property name="enabled" >
   <bool>true</bool>
  </property>
  <property name="geometry" >
   <rect>
    <x>0</x>
    <y>0</y>
    <width>555</width>
    <height>578</height>
   </rect>
  </property>
  <property name="windowTitle" >
   <string>Run Options</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" >
   <item>
    <widget class="QGroupBox" name="groupBox_3" >
     <property name="title" >
      <string>Executable</string>
     </property>
     <layout class="QHBoxLayout" name="horizontalLayout" >
      <item>
       <widget class="QRadioButton" name="targetRadio" >
        <property name="text" >
         <string>Target</string>
        </property>
        <property name="checked" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QRadioButton" name="execRadio" >
        <property name="text" >
         <string>Executable</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QWidget" native="1" name="executableWidget" >
        <property name="enabled" >
         <bool>true</bool>
        </property>
        <layout class="QHBoxLayout" name="horizontalLayout_2" >
         <item>
          <widget class="KUrlRequester" name="kcfg_executable" >
           <property name="clickMessage" >
            <string>Enter the target path here</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QWidget" native="1" name="widget_2" >
        <layout class="QHBoxLayout" name="horizontalLayout_3" >
         <item>
          <widget class="ProjectItemLineEdit" name="kcfg_runItem" >
           <property name="clickMessage" >
            <string>Enter the target here...</string>
           </property>
           <property name="showClearButton" stdset="0" >
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox" >
     <property name="title" >
      <string>Behaviour</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" >
      <item row="1" column="0" >
       <widget class="QLabel" name="mainprogram_label_3_2" >
        <property name="text" >
         <string>&amp;Working Directory:</string>
        </property>
        <property name="wordWrap" >
         <bool>false</bool>
        </property>
        <property name="buddy" >
         <cstring>kcfg_workingDirectory</cstring>
        </property>
       </widget>
      </item>
      <item row="1" column="1" >
       <widget class="KUrlRequester" name="kcfg_workingDirectory" />
      </item>
      <item row="2" column="0" >
       <widget class="QLabel" name="label" >
        <property name="text" >
         <string>&amp;Environment:</string>
        </property>
        <property name="buddy" >
         <cstring>kcfg_environment</cstring>
        </property>
       </widget>
      </item>
      <item row="2" column="1" >
       <widget class="KDevelop::EnvironmentSelectionWidget" name="kcfg_environment" />
      </item>
      <item row="3" column="1" >
       <widget class="QCheckBox" name="kcfg_startinterminal" >
        <property name="toolTip" >
         <string>Start the main program in an external terminal</string>
        </property>
        <property name="text" >
         <string/>
        </property>
       </widget>
      </item>
      <item row="3" column="0" >
       <widget class="QLabel" name="label_4" >
        <property name="text" >
         <string>Start in external terminal:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0" >
       <widget class="QLabel" name="mainprogram_label_2" >
        <property name="text" >
         <string>Arg&amp;uments:</string>
        </property>
        <property name="wordWrap" >
         <bool>false</bool>
        </property>
        <property name="buddy" >
         <cstring>kcfg_arguments</cstring>
        </property>
       </widget>
      </item>
      <item row="0" column="1" >
       <widget class="KLineEdit" name="kcfg_arguments" >
        <property name="toolTip" >
         <string>The command line arguments passed to the main program when run</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2" >
     <property name="title" >
      <string>Dependencies</string>
     </property>
     <layout class="QGridLayout" name="gridLayout" >
      <item row="0" column="0" >
       <widget class="QLabel" name="label_2" >
        <property name="text" >
         <string>Targets:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1" >
       <widget class="ProjectItemLineEdit" name="compilationProjectItem" >
        <property name="clickMessage" >
         <string>Enter the compilation dependency you need here...</string>
        </property>
        <property name="showClearButton" stdset="0" >
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="2" >
       <widget class="QPushButton" name="addCompilationProjectItem" >
        <property name="enabled" >
         <bool>false</bool>
        </property>
        <property name="text" >
         <string>+</string>
        </property>
       </widget>
      </item>
      <item rowspan="4" row="1" column="1" >
       <widget class="QListWidget" name="compileItems" >
        <property name="sizePolicy" >
         <sizepolicy vsizetype="Expanding" hsizetype="Expanding" >
          <horstretch>0</horstretch>
          <verstretch>2</verstretch>
         </sizepolicy>
        </property>
       </widget>
      </item>
      <item row="1" column="2" >
       <widget class="QPushButton" name="removeCompilationProjectItem" >
        <property name="enabled" >
         <bool>false</bool>
        </property>
        <property name="text" >
         <string>-</string>
        </property>
       </widget>
      </item>
      <item row="2" column="2" >
       <widget class="QPushButton" name="upItem" >
        <property name="enabled" >
         <bool>false</bool>
        </property>
        <property name="text" >
         <string>Up</string>
        </property>
       </widget>
      </item>
      <item row="3" column="2" >
       <widget class="QPushButton" name="downItem" >
        <property name="enabled" >
         <bool>false</bool>
        </property>
        <property name="text" >
         <string>Down</string>
        </property>
       </widget>
      </item>
      <item row="4" column="2" >
       <spacer name="verticalSpacer" >
        <property name="orientation" >
         <enum>Qt::Vertical</enum>
        </property>
        <property name="sizeHint" stdset="0" >
         <size>
          <width>20</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item row="5" column="0" >
       <widget class="QLabel" name="label_3" >
        <property name="text" >
         <string>Before Execution:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1" >
       <widget class="QComboBox" name="kcfg_beforeExec" >
        <property name="currentIndex" >
         <number>1</number>
        </property>
        <item>
         <property name="text" >
          <string>Do Nothing</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Build</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Build &amp; Install</string>
         </property>
        </item>
        <item>
         <property name="text" >
          <string>Build &amp; Install (as Super User)</string>
         </property>
        </item>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer_2" >
     <property name="orientation" >
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0" >
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11" />
 <layoutfunction spacing="KDialog::spacingHint" margin="KDialog::marginHint" />
 <pixmapfunction>qPixmapFromMimeSource</pixmapfunction>
 <customwidgets>
  <customwidget>
   <class>KLineEdit</class>
   <extends>QLineEdit</extends>
   <header>klineedit.h</header>
  </customwidget>
  <customwidget>
   <class>KUrlRequester</class>
   <extends>QFrame</extends>
   <header>kurlrequester.h</header>
  </customwidget>
  <customwidget>
   <class>KDevelop::EnvironmentSelectionWidget</class>
   <extends>KComboBox</extends>
   <header>util/environmentselectionwidget.h</header>
  </customwidget>
  <customwidget>
   <class>ProjectItemLineEdit</class>
   <extends>KLineEdit</extends>
   <header>runpreferences/projectitemlineedit.h</header>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>kcfg_arguments</tabstop>
  <tabstop>kcfg_workingDirectory</tabstop>
  <tabstop>kcfg_environment</tabstop>
 </tabstops>
 <includes>
  <include location="global" >ksqueezedtextlabel.h</include>
  <include location="local" >klineedit.h</include>
  <include location="local" >kurlrequester.h</include>
  <include location="local" >klineedit.h</include>
  <include location="local" >kpushbutton.h</include>
  <include location="local" >klineedit.h</include>
  <include location="local" >kurlrequester.h</include>
  <include location="local" >klineedit.h</include>
  <include location="local" >kpushbutton.h</include>
 </includes>
 <resources/>
 <connections>
  <connection>
   <sender>targetRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>widget_2</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>46</x>
     <y>59</y>
    </hint>
    <hint type="destinationlabel" >
     <x>383</x>
     <y>78</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>execRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>executableWidget</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>153</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel" >
     <x>196</x>
     <y>76</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>targetRadio</sender>
   <signal>clicked()</signal>
   <receiver>kcfg_executable</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>72</x>
     <y>51</y>
    </hint>
    <hint type="destinationlabel" >
     <x>221</x>
     <y>52</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>execRadio</sender>
   <signal>clicked()</signal>
   <receiver>kcfg_runItem</receiver>
   <slot>clear()</slot>
   <hints>
    <hint type="sourcelabel" >
     <x>157</x>
     <y>62</y>
    </hint>
    <hint type="destinationlabel" >
     <x>418</x>
     <y>62</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
